Remotion/Validation/UnitTests/ValidationRuleCollectorBaseTest.cs

```diff
@@ -74,6 +77,33 @@ public void AddRule_ForProperty_OneRule ()
       Assert.That(resultAsComponentRuleBuilder.PropertyMetaValidationRuleCollector, Is.SameAs(metaValidationPropertyRule));
     }
 
+    [Test]
+    public void AddRule_ForPropertyViaPropertyInformation_OneRule ()
+    {
+      var propertyInfo = PropertyInfoAdapter.Create(MemberInfoFromExpressionUtility.GetProperty(_firstNameExpression));
+      Func<object,object> propertyGetter = o => ((Customer)o).FirstName;
+      var result = _customerValidationRuleCollector.AddRule<string>(propertyInfo, propertyGetter);
+
+      Assert.That(_customerValidationRuleCollector.AddedPropertyRules.Count, Is.EqualTo(1));
+      Assert.That(_customerValidationRuleCollector.PropertyMetaValidationRules.Count, Is.EqualTo(1));
+      Assert.That(_customerValidationRuleCollector.RemovedPropertyRules.Count, Is.EqualTo(0));
+
+      var propertyRule = _customerValidationRuleCollector.AddedPropertyRules.First();
+      var metaValidationPropertyRule = _customerValidationRuleCollector.PropertyMetaValidationRules.First();
+
+      Assert.That(result, Is.TypeOf(typeof(AddingPropertyValidationRuleBuilder<Customer, string>)));
+      var resultAsComponentRuleBuilder = (AddingPropertyValidationRuleBuilder<Customer, string>)result;
+      Assert.That(resultAsComponentRuleBuilder.AddingPropertyValidationRuleCollector.CollectorType, Is.SameAs(_customerValidationRuleCollector.GetType()));
+      Assert.That(resultAsComponentRuleBuilder.AddingPropertyValidationRuleCollector.Property, Is.SameAs(propertyInfo));
+      Assert.That(resultAsComponentRuleBuilder.AddingPropertyValidationRuleCollector, Is.SameAs(propertyRule));
+      Assert.That(resultAsComponentRuleBuilder.AddingPropertyValidationRuleCollector.Validators.Count(), Is.EqualTo(0));
+      Assert.That(resultAsComponentRuleBuilder.PropertyMetaValidationRuleCollector.CollectorType, Is.SameAs(_customerValidationRuleCollector.GetType()));
+      Assert.That(resultAsComponentRuleBuilder.PropertyMetaValidationRuleCollector, Is.SameAs(metaValidationPropertyRule));
+      var resultCollector = (AddingPropertyValidationRuleCollector<Customer, string>)resultAsComponentRuleBuilder.AddingPropertyValidationRuleCollector;
+      Assert.That(resultCollector.Property, Is.SameAs(propertyInfo));
+      Assert.That(resultCollector.PropertyFunc, Is.SameAs(propertyGetter));
```

Comment:
```suggestion
      var resultCollector = (AddingPropertyValidationRuleCollector<Customer, string>)resultAsComponentRuleBuilder.AddingPropertyValidationRuleCollector;
      Assert.That(resultCollector.PropertyFunc, Is.SameAs(propertyGetter));

      Assert.That(resultAsComponentRuleBuilder.PropertyMetaValidationRuleCollector.CollectorType, Is.SameAs(_customerValidationRuleCollector.GetType()));
      Assert.That(resultAsComponentRuleBuilder.PropertyMetaValidationRuleCollector, Is.SameAs(metaValidationPropertyRule));
```

Reformatting
Dropped the line with the propertyInfo-test, it's redundant given line 97